### DISCUSSION TITLE
Integration with Python

### DISCUSSION THREAD

**Discussion by simran-suri** [AbsaOSS/spline-spark-agent#492] [Q&A]
Hi Team,

I'm trying to create lineage for my pipelines and for it I used sc._jvm.za.co.absa.spline.harvester.SparkLineageInitializer.enableLineageTracking(spark._jsparkSession) above my python code. 

Now, I need to install spark-3.1-spline-agent-bundle_2.12:0.7.8 in my python code repository, to create that lineage. 

Is there any way to install this maven dependency in python code?

**Comment by wajda** [comment]
What exactly do you mean?
The capability of downloading Maven artifacts depends on the tool/platform that you are using, not the programming language.
For example, with the `pyspark` you can do `pyspark --packages za.co.absa.spline.agent.spark:spark-3.1-spline-agent-bundle_2.12:0.7.8` while the Databricks notebook provides a specific UI for installing packages.

**Reply by simran-suri** [reply]
Error:

![image](https://user-images.githubusercontent.com/101385030/188566789-24450b4c-9792-4993-b712-bf1d46ea345d.png)


Code:
from pyspark import SparkContext
from pyspark.sql import SparkSession

spark = SparkSession.builder \
    .config("spark.jars", "jar_file/spark-2.4-spline-agent-bundle_2.12-0.7.10.jar")\
    .getOrCreate()

sc = spark.sparkContext
sc.setSystemProperty("spark.jars", "jar_file/spark-2.4-spline-agent-bundle_2.12-0.7.10.jar")
sc.setSystemProperty("spark._jsparkSession","za.co.absa.spline.harvester.SparkLineageInitializer.enableLineageTracking")

jvm = sc._jvm
sc._jvm.za.co.absa.spline.harvester.SparkLineageInitializer.enableLineageTracking(spark._jsparkSession)

**Reply by simran-suri** [reply]
Hi @wajda 

Can you please look into the code and suggest what can be done to resolve this error?

**Reply by wajda** [reply]
The code seems to be OK, make sure you are using Spline agent bundle compatible to the Spark and Scala version in use.

**Reply by simran-suri** [reply]
Yes, the version are compatible with spline agent bundle, but not able to figure out this error.

**Comment by wajda** [comment]
The error means that Spline cannot find a `org.apache.spark.internal.Logging` constructor with an expected signature. It could be caused by the fact that you are using another Spark version than 2.4 (which is what your Spline bundle compiled against), or it could be a proprietary (modified) Spark fork.

What exactly are you using?

**Reply by simran-suri** [reply]
pyspark==2.4.4

I'm using this in my requirement file as pip install

**Comment by wajda** [comment]
Try to install Spark directly, not via pip
